pkgdb/src/buildenv/assets/activate.d/tcsh

```diff
@@ -0,0 +1,47 @@
+setenv _gnused "@gnused@"
+
+if ( $_flox_activate_tracelevel >= 2 ) then
+  set verbose
+endif
+
+# Assert that the expected _{add,del}_env variables are present.
+if ( ! $?_add_env || ! $?_del_env ) then
+  echo 'ERROR (tcsh): $_add_env and $_del_env not found in environment' >&2
+  exit 1
+endif
+
+# We invoke tcsh with -f to prevent the user and system-wide
+# config files from coming back and overriding the environment
+# that we're trying to set up here, so we start by sourcing
+# these files ourselves prior to setting up the environment.
+foreach f ( /etc/csh.login /etc/csh.cshrc ~/.cshrc ~/.tcshrc )
+  if ( -f $f ) then
+    source $f
```

Comment:
question blocking: could we hit the same infinite looping bug as with .bashrc?

Reply:
We have the `activate:infinite_source:{fish,tcsh}` tests for that, and they pass so I'm confident we're OK in that regard. Or at least I am following commit 8c0eddd2a48f8134c7683b14e01c37568392a6d7 in which I tweaked the generation of the "*.extra" dotfiles to make sure their contents were 100% correct. I'm not sure when this regressed but they weren't doing much of anything previously (I suspect bats may have been involved...)  👍

Reply:
I'm going to go ahead and hit resolve conversation, but let me know if you have any other questions.